widget/list.go

```diff
@@ -91,7 +91,11 @@ func newListRenderer(objects []fyne.CanvasObject, l *List, scroller *ScrollConta
 }
 
 func (l *listRenderer) Layout(size fyne.Size) {
-	if l.list.Length() == 0 {
+	lengthFunc := l.list.Length
+	if lengthFunc == nil {
+		return
+	}
+	if lengthFunc() == 0 {
```

Comment:
nit: Length is called multiple times
```suggestion
	length := 0
	if f == l.list.Length; f != nil {
		length = f()
	}
	if length == 0 {
		return
	}
```

Reply:
resolved